diff --git a/src/print_mpd.c b/src/print_mpd.c
index 9a77a14..4df72ff 100644
--- a/src/print_mpd.c
+++ b/src/print_mpd.c
@@ -96,17 +96,17 @@ void print_mpd(yajl_gen json_gen, char* buffer,
     struct mpd_status* status = 0;
     struct mpd_song* song = 0;
 
-    if (co != 0 && mpd_connection_get_error(co) != MPD_ERROR_SUCCESS) {
+    if (co && mpd_connection_get_error(co) != MPD_ERROR_SUCCESS && !mpd_connection_clear_error(co)) {
         mpd_connection_free(co);
         co = 0;
     }
 
-    if (co == 0) {
+    if (!co) {
         co = mpd_connection_new(host, port, 0);
-        if (co != 0 && password != 0 && *password != 0)  // if password not empty
+        if (co && password && *password)  // if password not empty
             mpd_run_password(co, password);
         int err = mpd_connection_get_error(co);
-        if (co == 0 || err != MPD_ERROR_SUCCESS) {
+        if (!co || err != MPD_ERROR_SUCCESS) {
             START_COLOR("color_bad");
             outwalk += sprintf(buffer, "%s", format_off);
 #ifdef DEBUG
@@ -179,12 +179,6 @@ end:
         mpd_status_free(status);
     if (song)
         mpd_song_free(song);
-    if (co) {
-        if (mpd_connection_get_error(co) != MPD_ERROR_SUCCESS) {
-            mpd_connection_free(co);
-            co = 0;
-        }
-    }
 
     END_COLOR;
     OUTPUT_FULL_TEXT(buffer);
