From e8bd8dfbf19985ea5f9267370065a555a9c9aeca Mon Sep 17 00:00:00 2001
From: Szabolcs Nagy <nsz@port70.net>
Date: Thu, 5 Nov 2015 23:48:11 +0000
Subject: [PATCH 14/19] powerpc default secureplt

---
 gcc/config.gcc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index 16ee9ab..046121c 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -2416,6 +2416,10 @@ powerpc*-*-linux*)
 	    powerpc*-*-linux*paired*)
 		tm_file="${tm_file} rs6000/750cl.h" ;;
 	esac
+	case ${target} in
+	    *-linux*-musl*)
+		enable_secureplt=yes ;;
+	esac
 	if test x${enable_secureplt} = xyes; then
 		tm_file="rs6000/secureplt.h ${tm_file}"
 	fi
-- 
2.4.1

