--- zig-9999/cmake/Findclang.cmake.orig
+++ zig-9999/cmake/Findclang.cmake
@@ -8,6 +8,7 @@
 
 find_path(CLANG_INCLUDE_DIRS NAMES clang/Frontend/ASTUnit.h
     PATHS
+        /usr/lib/llvm/4/include
         /usr/lib/llvm-4.0/include
         /mingw64/include)
 
@@ -15,6 +16,7 @@ macro(FIND_AND_ADD_CLANG_LIB _libname_)
     string(TOUPPER ${_libname_} _prettylibname_)
     find_library(CLANG_${_prettylibname_}_LIB NAMES ${_libname_}
         PATHS
+            /usr/lib/llvm/4/lib
             /usr/lib/llvm-4.0/lib
             /mingw64/lib)
     if(CLANG_${_prettylibname_}_LIB)
