diff --git opensmtpd-5.9.2p1/openbsd-compat/openbsd-compat.h opensmtpd-5.9.2p1/openbsd-compat/openbsd-compat.h
--- opensmtpd-5.9.2p1/openbsd-compat/openbsd-compat.h
+++ opensmtpd-5.9.2p1/openbsd-compat/openbsd-compat.h
@@ -208,6 +208,16 @@
 void errc(int, int, const char *, ...);
 #endif
 
+#ifndef HAVE_INET_NET_PTON
+unsigned int arc4random(void);
+void arc4random_buf(void *, size_t);
+uint32_t arc4random_uniform(uint32_t);
+void explicit_bzero(void *p, size_t n);
+int inet_net_pton(int, const char *, void *, size_t);
+void *reallocarray(void *, size_t, size_t);
+int res_hnok(const char *dn);
+#endif
+
 #ifndef HAVE_PLEDGE
 #define pledge(promises, paths) 0
 #endif
diff --git opensmtpd-6.0.2p1/smtpd/smtpd.c opensmtpd-fixed/smtpd/smtpd.c
index 6cd270f..05c46e1 100644
--- opensmtpd-6.0.2p1/smtpd/smtpd.c
+++ opensmtpd-fixed/smtpd/smtpd.c
@@ -326,7 +326,7 @@ parent_shutdown(int ret)
 			kill(child->pid, SIGTERM);
 
 	do {
-		pid = waitpid(WAIT_MYPGRP, NULL, 0);
+		pid = waitpid(0, NULL, 0);
 	} while (pid != -1 || (pid == -1 && errno == EINTR));
 
 	unlink(SMTPD_SOCKET);
