--- a/net/bindu.c
+++ b/net/bindu.c
@@ -22,6 +22,7 @@
 #include <sys/un.h>
 #include <unistd.h>
 #include "socket.h"
+#include <string.h>
 
 /** Bind a UNIX domain address (path) to a socket. */
 int socket_bindu(int sock, const char* path)
--- a/net/connectu.c
+++ b/net/connectu.c
@@ -22,6 +22,7 @@
 #include <sys/un.h>
 #include <unistd.h>
 #include "socket.h"
+#include <string.h>
 
 /** Make an UNIX domain connection. */
 int socket_connectu(int sock, const char* path)
