diff --git chromium-81.0.4044.113/third_party/blink/renderer/core/editing/commands/delete_selection_command.cc chromium-81.0.4044.113/third_party/blink/renderer/core/editing/commands/delete_selection_command.cc
index 62204351fd..e73dc86ef9 100644
--- chromium-81.0.4044.113/third_party/blink/renderer/core/editing/commands/delete_selection_command.cc
+++ chromium-81.0.4044.113/third_party/blink/renderer/core/editing/commands/delete_selection_command.cc
@@ -225,6 +225,10 @@ static Position TrailingWhitespacePosition(const Position& position,
   return Position();
 }
 
+static bool IsAHTMLTableRowElement(const Node* node) {
+  return IsA<HTMLTableRowElement>(node);
+}
+
 void DeleteSelectionCommand::InitializePositionData(
     EditingState* editing_state) {
   DCHECK(!GetDocument().NeedsLayoutTreeUpdate());
@@ -254,9 +258,9 @@ void DeleteSelectionCommand::InitializePositionData(
   end_root_ = RootEditableElementOf(end);
 
   start_table_row_ = To<HTMLTableRowElement>(
-      EnclosingNodeOfType(start, &IsA<HTMLTableRowElement>));
+      EnclosingNodeOfType(start, &IsAHTMLTableRowElement));
   end_table_row_ = To<HTMLTableRowElement>(
-      EnclosingNodeOfType(end, &IsA<HTMLTableRowElement>));
+      EnclosingNodeOfType(end, &IsAHTMLTableRowElement));
 
   // Don't move content out of a table cell.
   // If the cell is non-editable, enclosingNodeOfType won't return it by
