diff --git chromium-85.0.4183.83/chrome/browser/chrome_browser_main_linux.cc chromium-85.0.4183.83/chrome/browser/chrome_browser_main_linux.cc
index f6ce26c769..271ac75b9d 100644
--- chromium-85.0.4183.83/chrome/browser/chrome_browser_main_linux.cc
+++ chromium-85.0.4183.83/chrome/browser/chrome_browser_main_linux.cc
@@ -101,7 +101,7 @@ void ChromeBrowserMainPartsLinux::PostProfileInit() {
 }
 
 void ChromeBrowserMainPartsLinux::PostMainMessageLoopStart() {
-#if !defined(OS_CHROMEOS)
+#if !defined(OS_CHROMEOS) && defined(USE_DBUS)
   bluez::BluezDBusManager::Initialize(nullptr /* system_bus */);
 #endif
 
@@ -109,7 +109,7 @@ void ChromeBrowserMainPartsLinux::PostMainMessageLoopStart() {
 }
 
 void ChromeBrowserMainPartsLinux::PostDestroyThreads() {
-#if !defined(OS_CHROMEOS)
+#if !defined(OS_CHROMEOS) && defined(USE_DBUS)
   bluez::BluezDBusManager::Shutdown();
   bluez::BluezDBusThreadManager::Shutdown();
 #endif
diff --git chromium-85.0.4183.83/chrome/browser/metrics/bluetooth_available_utility.cc chromium-85.0.4183.83/chrome/browser/metrics/bluetooth_available_utility.cc
index 8acdf7c985..c626975091 100644
--- chromium-85.0.4183.83/chrome/browser/metrics/bluetooth_available_utility.cc
+++ chromium-85.0.4183.83/chrome/browser/metrics/bluetooth_available_utility.cc
@@ -13,7 +13,7 @@
 #include "device/bluetooth/bluetooth_adapter.h"
 #include "device/bluetooth/bluetooth_adapter_factory.h"
 
-#if defined(OS_LINUX)
+#if defined(OS_LINUX) && defined(USE_DBUS)
 #include "device/bluetooth/dbus/bluez_dbus_manager.h"
 #endif  // defined(OS_LINUX)
 
@@ -55,7 +55,7 @@ void ReportBluetoothAvailability() {
     return;
   }
 
-#if defined(OS_LINUX)
+#if defined(OS_LINUX) && defined(USE_DBUS)
   // This is for tests that have not initialized bluez or dbus thread manager.
   // Outside of tests these are initialized earlier during browser startup.
   if (!bluez::BluezDBusManager::IsInitialized())
diff --git chromium-85.0.4183.83/chrome/browser/notifications/notification_display_service_impl.cc chromium-85.0.4183.83/chrome/browser/notifications/notification_display_service_impl.cc
index 85d20f180a..61d2135949 100644
--- chromium-85.0.4183.83/chrome/browser/notifications/notification_display_service_impl.cc
+++ chromium-85.0.4183.83/chrome/browser/notifications/notification_display_service_impl.cc
@@ -47,6 +47,7 @@
 
 namespace {
 
+#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
 #if !defined(OS_CHROMEOS)
 bool NativeNotificationsEnabled(Profile* profile) {
 #if defined(OS_LINUX)
@@ -60,6 +61,7 @@ bool NativeNotificationsEnabled(Profile* profile) {
   return base::FeatureList::IsEnabled(features::kNativeNotifications);
 }
 #endif
+#endif
 
 // Returns the NotificationPlatformBridge to use for the current platform.
 // Will return a nullptr for platforms that don't support native notifications.
diff --git chromium-85.0.4183.83/chrome/browser/platform_util_linux.cc chromium-85.0.4183.83/chrome/browser/platform_util_linux.cc
index 86cb40c333..c2810d8100 100644
--- chromium-85.0.4183.83/chrome/browser/platform_util_linux.cc
+++ chromium-85.0.4183.83/chrome/browser/platform_util_linux.cc
@@ -13,14 +13,18 @@
 #include "base/threading/scoped_blocking_call.h"
 #include "chrome/browser/chrome_notification_types.h"
 #include "chrome/browser/platform_util_internal.h"
+#if defined(USE_DBUS)
 #include "components/dbus/thread_linux/dbus_thread_linux.h"
+#endif
 #include "content/public/browser/browser_thread.h"
 #include "content/public/browser/notification_observer.h"
 #include "content/public/browser/notification_registrar.h"
 #include "content/public/browser/notification_service.h"
+#if defined(USE_DBUS)
 #include "dbus/bus.h"
 #include "dbus/message.h"
 #include "dbus/object_proxy.h"
+#endif
 #include "url/gurl.h"
 
 using content::BrowserThread;
@@ -29,6 +33,7 @@ namespace platform_util {
 
 namespace {
 
+#if defined(USE_DBUS)
 const char kFreedesktopFileManagerName[] = "org.freedesktop.FileManager1";
 const char kFreedesktopFileManagerPath[] = "/org/freedesktop/FileManager1";
 
@@ -111,6 +116,7 @@ class ShowItemHelper : public content::NotificationObserver {
 
   base::WeakPtrFactory<ShowItemHelper> weak_ptr_factory_{this};
 };
+#endif
 
 void RunCommand(const std::string& command,
                 const base::FilePath& working_directory,
@@ -177,8 +183,10 @@ void PlatformOpenVerifiedItem(const base::FilePath& path, OpenItemType type) {
 }  // namespace internal
 
 void ShowItemInFolder(Profile* profile, const base::FilePath& full_path) {
+#if defined(USE_DBUS)
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
   ShowItemHelper::GetInstance().ShowItemInFolder(profile, full_path);
+#endif
 }
 
 void OpenExternal(Profile* profile, const GURL& url) {
diff --git chromium-85.0.4183.83/chrome/browser/ui/BUILD.gn chromium-85.0.4183.83/chrome/browser/ui/BUILD.gn
index a429fd9751..d60d804755 100644
--- chromium-85.0.4183.83/chrome/browser/ui/BUILD.gn
+++ chromium-85.0.4183.83/chrome/browser/ui/BUILD.gn
@@ -3065,10 +3065,6 @@ static_library("ui") {
 
     if (use_x11) {
       sources += [
-        "views/frame/global_menu_bar_registrar_x11.cc",
-        "views/frame/global_menu_bar_registrar_x11.h",
-        "views/frame/global_menu_bar_x11.cc",
-        "views/frame/global_menu_bar_x11.h",
         "views/frame/native_browser_frame_factory_aurax11.cc",
         "views/javascript_app_modal_dialog_views_x11.cc",
         "views/javascript_app_modal_dialog_views_x11.h",
@@ -3076,11 +3072,21 @@ static_library("ui") {
         "views/javascript_app_modal_event_blocker_x11.h",
       ]
       deps += [
-        "//components/dbus/menu",
         "//ui/events/devices",
         "//ui/events/devices/x11",
         "//ui/events/platform/x11",
       ]
+      if (use_dbus) {
+        sources += [
+          "views/frame/global_menu_bar_registrar_x11.cc",
+          "views/frame/global_menu_bar_registrar_x11.h",
+          "views/frame/global_menu_bar_x11.cc",
+          "views/frame/global_menu_bar_x11.h",
+        ]
+        deps += [
+          "//components/dbus/menu",
+        ]
+      }
     }
     if (use_ozone) {
       if (!is_desktop_linux) {
diff --git chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc
index 11eb4519eb..c86a4f63ca 100644
--- chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc
+++ chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.cc
@@ -76,7 +76,7 @@ void BrowserDesktopWindowTreeHostLinux::Init(
     const views::Widget::InitParams& params) {
   DesktopWindowTreeHostLinuxImpl::Init(std::move(params));
 
-#if defined(USE_X11)
+#if defined(USE_X11) && defined(USE_DBUS)
   // We have now created our backing X11 window. We now need to (possibly)
   // alert Unity that there's a menu bar attached to it.
   if (!features::IsUsingOzonePlatform())
@@ -86,7 +86,7 @@ void BrowserDesktopWindowTreeHostLinux::Init(
 }
 
 void BrowserDesktopWindowTreeHostLinux::CloseNow() {
-#if defined(USE_X11)
+#if defined(USE_X11) && defined(USE_DBUS)
   global_menu_bar_x11_.reset();
 #endif
   DesktopWindowTreeHostLinuxImpl::CloseNow();
diff --git chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.h chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.h
index 6077cb0e65..4e3eb1f981 100644
--- chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.h
+++ chromium-85.0.4183.83/chrome/browser/ui/views/frame/browser_desktop_window_tree_host_linux.h
@@ -9,7 +9,9 @@
 #include "chrome/browser/ui/views/frame/browser_desktop_window_tree_host.h"
 
 #if defined(USE_X11)
+#if defined(USE_DBUS)
 #include "chrome/browser/ui/views/frame/global_menu_bar_x11.h"  // nogncheck
+#endif
 #include "ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h"  // nogncheck
 #else
 #include "ui/views/widget/desktop_aura/desktop_window_tree_host_linux.h"  // nogncheck
@@ -63,7 +65,7 @@ class BrowserDesktopWindowTreeHostLinux
   BrowserView* browser_view_ = nullptr;
   BrowserFrame* browser_frame_ = nullptr;
 
-#if defined(USE_X11)
+#if defined(USE_X11) && defined(USE_DBUS)
   // Each browser frame maintains its own menu bar object because the lower
   // level dbus protocol associates a xid to a menu bar; we can't map multiple
   // xids to the same menu bar.
diff --git chromium-85.0.4183.83/device/bluetooth/bluetooth_adapter_factory.cc chromium-85.0.4183.83/device/bluetooth/bluetooth_adapter_factory.cc
index f39209367a..94267acf3a 100644
--- chromium-85.0.4183.83/device/bluetooth/bluetooth_adapter_factory.cc
+++ chromium-85.0.4183.83/device/bluetooth/bluetooth_adapter_factory.cc
@@ -45,7 +45,7 @@ bool BluetoothAdapterFactory::IsBluetoothSupported() {
   // instance even on platforms that would otherwise not support it.
   if (Get()->adapter_)
     return true;
-#if defined(OS_ANDROID) || defined(OS_WIN) || defined(OS_LINUX) || \
+#if defined(OS_ANDROID) || defined(OS_WIN) || (defined(OS_LINUX) && defined(USE_DBUS)) || \
     defined(OS_MACOSX)
   return true;
 #else
@@ -68,7 +68,7 @@ bool BluetoothAdapterFactory::IsLowEnergySupported() {
   return base::win::GetVersion() >= base::win::Version::WIN10;
 #elif defined(OS_MACOSX)
   return true;
-#elif defined(OS_LINUX)
+#elif defined(OS_LINUX) && defined(USE_DBUS)
   return true;
 #else
   return false;
diff --git chromium-85.0.4183.83/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc chromium-85.0.4183.83/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
index 2d4cf40561..3ec27fe8bc 100644
--- chromium-85.0.4183.83/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
+++ chromium-85.0.4183.83/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
@@ -1328,7 +1328,7 @@ void BluetoothLowEnergyCreateServiceFunction::DoWork() {
 // TODO: Ideally this should be handled by our feature system, so that this
 // code doesn't even compile on OSes it isn't being used on, but currently this
 // is not possible.
-#if !defined(OS_WIN)
+#if !defined(OS_WIN) && (!defined(OS_LINUX) || defined(USE_DBUS))
   base::WeakPtr<device::BluetoothLocalGattService> service =
       device::BluetoothLocalGattService::Create(
           event_router_->adapter(),
@@ -1357,6 +1357,7 @@ bool BluetoothLowEnergyCreateCharacteristicFunction::ParseParams() {
 }
 
 void BluetoothLowEnergyCreateCharacteristicFunction::DoWork() {
+#if !defined(OS_LINUX) || defined(USE_DBUS)
   device::BluetoothLocalGattService* service =
       event_router_->adapter()->GetGattService(params_->service_id);
   if (!service) {
@@ -1377,6 +1378,9 @@ void BluetoothLowEnergyCreateCharacteristicFunction::DoWork() {
 
   Respond(ArgumentList(apibtle::CreateCharacteristic::Results::Create(
       characteristic->GetIdentifier())));
+#else
+  Respond(Error(kErrorPlatformNotSupported));
+#endif
 }
 
 // createDescriptor:
@@ -1393,6 +1397,7 @@ bool BluetoothLowEnergyCreateDescriptorFunction::ParseParams() {
 }
 
 void BluetoothLowEnergyCreateDescriptorFunction::DoWork() {
+#if !defined(OS_LINUX) || defined(USE_DBUS)
   device::BluetoothLocalGattCharacteristic* characteristic =
       event_router_->GetLocalCharacteristic(params_->characteristic_id);
   if (!characteristic) {
@@ -1408,6 +1413,9 @@ void BluetoothLowEnergyCreateDescriptorFunction::DoWork() {
 
   Respond(ArgumentList(
       apibtle::CreateDescriptor::Results::Create(descriptor->GetIdentifier())));
+#else
+  Respond(Error(kErrorPlatformNotSupported));
+#endif
 }
 
 // registerService:
diff --git chromium-85.0.4183.83/services/device/battery/battery_status_manager_default.cc chromium-85.0.4183.83/services/device/battery/battery_status_manager_default.cc
index 35268761d0..dbaa8722c6 100644
--- chromium-85.0.4183.83/services/device/battery/battery_status_manager_default.cc
+++ chromium-85.0.4183.83/services/device/battery/battery_status_manager_default.cc
@@ -22,11 +22,10 @@ class BatteryStatusManagerDefault : public BatteryStatusManager {
  private:
   // BatteryStatusManager:
   bool StartListeningBatteryChange() override {
-    NOTIMPLEMENTED();
     return false;
   }
 
-  void StopListeningBatteryChange() override { NOTIMPLEMENTED(); }
+  void StopListeningBatteryChange() override { }
 
   DISALLOW_COPY_AND_ASSIGN(BatteryStatusManagerDefault);
 };
diff --git chromium-85.0.4183.83/services/device/wake_lock/power_save_blocker/BUILD.gn chromium-85.0.4183.83/services/device/wake_lock/power_save_blocker/BUILD.gn
index 9365bd609d..fea5949241 100644
--- chromium-85.0.4183.83/services/device/wake_lock/power_save_blocker/BUILD.gn
+++ chromium-85.0.4183.83/services/device/wake_lock/power_save_blocker/BUILD.gn
@@ -70,6 +70,9 @@ if (is_android) {
     visibility = [ ":power_save_blocker" ]
     sources = java_sources_needing_jni
   }
+  if (!is_chromeos && use_x11 && !use_dbus) {
+    sources += [ "power_save_blocker_ozone.cc" ]
+  }
 
   android_library("java") {
     # Conceptually, this should be visible only to //services/device:java.
