diff --git chromium-65.0.3325.146/chrome/browser/notifications/notification_display_service_impl.cc
index cf8fb8b..4ebd523 100644
--- chromium-65.0.3325.146/chrome/browser/notifications/notification_display_service_impl.cc.orig
+++ chromium-65.0.3325.146/chrome/browser/notifications/notification_display_service_impl.cc
@@ -212,9 +212,13 @@ void NotificationDisplayServiceImpl::Display(
   }
 
   NotificationPlatformBridge* bridge =
+#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
       NotificationPlatformBridge::CanHandleType(notification_type)
           ? bridge_
           : message_center_bridge_.get();
+#else
+          message_center_bridge_.get();
+#endif
   DCHECK(bridge);
 
   bridge->Display(notification_type, profile_, notification,
@@ -236,9 +240,13 @@ void NotificationDisplayServiceImpl::Close(
   }
 
   NotificationPlatformBridge* bridge =
+#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
       NotificationPlatformBridge::CanHandleType(notification_type)
           ? bridge_
           : message_center_bridge_.get();
+#else
+          message_center_bridge_.get();
+#endif
   DCHECK(bridge);
 
   bridge->Close(profile_, notification_id);
@@ -283,10 +291,12 @@ void NotificationDisplayServiceImpl::GetDisplayed(
 void NotificationDisplayServiceImpl::OnNotificationPlatformBridgeReady(
     bool success) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
   if (base::FeatureList::IsEnabled(features::kNativeNotifications)) {
     UMA_HISTOGRAM_BOOLEAN("Notifications.UsingNativeNotificationCenter",
                           success);
   }
+#endif
 
   if (!success) {
     // Fall back to the message center if initialization failed. Initialization
