diff --git chromium-65.0.3325.146/chrome/browser/ui/libgtkui/gtk_util.cc
index fc6ffb9..f1e0bd6 100644
--- chromium-65.0.3325.146/chrome/browser/ui/libgtkui/gtk_util.cc.orig
+++ chromium-65.0.3325.146/chrome/browser/ui/libgtkui/gtk_util.cc
@@ -56,6 +56,7 @@ void CommonInitFromCommandLine(const base::CommandLine& command_line,
   }
 }
 
+#if GTK_MAJOR_VERSION > 2
 using GtkSetState = void (*)(GtkWidgetPath*, gint, GtkStateFlags);
 PROTECTED_MEMORY_SECTION base::ProtectedMemory<GtkSetState>
     _gtk_widget_path_iter_set_state;
@@ -63,6 +64,7 @@ PROTECTED_MEMORY_SECTION base::ProtectedMemory<GtkSetState>
 using GtkSetObjectName = void (*)(GtkWidgetPath*, gint, const char*);
 PROTECTED_MEMORY_SECTION base::ProtectedMemory<GtkSetObjectName>
     _gtk_widget_path_iter_set_object_name;
+#endif
 
 }  // namespace
 
