diff --git chromium-60.0.3112.78/chrome/browser/ui/libgtkui/gtk_ui.cc chromium-60.0.3112.78/chrome/browser/ui/libgtkui/gtk_ui.cc
index 7df1d0c..c20ec08 100644
--- chromium-60.0.3112.78/chrome/browser/ui/libgtkui/gtk_ui.cc
+++ chromium-60.0.3112.78/chrome/browser/ui/libgtkui/gtk_ui.cc
@@ -315,6 +315,7 @@ gfx::FontRenderParams GetGtkFontRenderParams() {
   return params;
 }
 
+#if GTK_MAJOR_VERSION > 2
 float GetRawDeviceScaleFactor() {
   if (display::Display::HasForceDeviceScaleFactor())
     return display::Display::GetForcedDeviceScaleFactor();
@@ -325,6 +326,7 @@ float GetRawDeviceScaleFactor() {
   gdouble resolution = gdk_screen_get_resolution(screen);
   return resolution <= 0 ? scale : resolution * scale / kDefaultDPI;
 }
+#endif
 
 views::LinuxUI::NonClientMiddleClickAction GetDefaultMiddleClickAction() {
   std::unique_ptr<base::Environment> env(base::Environment::Create());
@@ -1038,6 +1040,9 @@ void GtkUi::ResetStyle() {
 }
 
 void GtkUi::UpdateDeviceScaleFactor() {
+#if GTK_MAJOR_VERSION == 2
+  device_scale_factor_ = 1.0f;
+#else
   // Note: Linux chrome currently does not support dynamic DPI
   // changes.  This is to allow flags to override the DPI settings
   // during startup.
@@ -1045,6 +1050,7 @@ void GtkUi::UpdateDeviceScaleFactor() {
   // Blacklist scaling factors <120% (crbug.com/484400) and round
   // to 1 decimal to prevent rendering problems (crbug.com/485183).
   device_scale_factor_ = scale < 1.2f ? 1.0f : roundf(scale * 10) / 10;
+#endif
   UpdateDefaultFont();
 }
 
