diff --git chromium-69.0.3497.100/chrome/browser/BUILD.gn
index b5d398f..2e53724 100644
--- chromium-69.0.3497.100/chrome/browser/BUILD.gn
+++ chromium-69.0.3497.100/chrome/browser/BUILD.gn
@@ -3215,13 +3215,17 @@ jumbo_split_static_library("browser") {
       "first_run/upgrade_util_linux.cc",
       "first_run/upgrade_util_linux.h",
       "icon_loader_auralinux.cc",
-      "password_manager/native_backend_kwallet_x.cc",
-      "password_manager/native_backend_kwallet_x.h",
       "platform_util_linux.cc",
       "shell_integration_linux.cc",
       "shell_integration_linux.h",
       "speech/tts_linux.cc",
     ]
+    if (use_dbus) {
+      sources += [
+        "password_manager/native_backend_kwallet_x.cc",
+        "password_manager/native_backend_kwallet_x.h",
+      ]
+    }
 
     deps += [ "//third_party/speech-dispatcher" ]
 
diff --git chromium-69.0.3497.100/chrome/browser/chrome_browser_main_linux.cc
index 3a26947..50c9452 100644
--- chromium-69.0.3497.100/chrome/browser/chrome_browser_main_linux.cc
+++ chromium-69.0.3497.100/chrome/browser/chrome_browser_main_linux.cc
@@ -93,7 +93,7 @@ void ChromeBrowserMainPartsLinux::PostProfileInit() {
 }
 
 void ChromeBrowserMainPartsLinux::PostMainMessageLoopStart() {
-#if !defined(OS_CHROMEOS)
+#if !defined(OS_CHROMEOS) && defined(USE_DBUS)
   bluez::BluezDBusThreadManager::Initialize();
   bluez::BluezDBusManager::Initialize();
 #endif
@@ -103,7 +103,7 @@ void ChromeBrowserMainPartsLinux::PostMainMessageLoopStart() {
 }
 
 void ChromeBrowserMainPartsLinux::PostDestroyThreads() {
-#if !defined(OS_CHROMEOS)
+#if !defined(OS_CHROMEOS) && defined(USE_DBUS)
   bluez::BluezDBusManager::Shutdown();
   bluez::BluezDBusThreadManager::Shutdown();
 #endif
diff --git chromium-69.0.3497.100/chrome/browser/password_manager/password_store_factory.cc
index a234d58..465853e 100644
--- chromium-69.0.3497.100/chrome/browser/password_manager/password_store_factory.cc
+++ chromium-69.0.3497.100/chrome/browser/password_manager/password_store_factory.cc
@@ -203,6 +203,7 @@ PasswordStoreFactory::BuildServiceInstanceFor(
   std::unique_ptr<PasswordStoreX::NativeBackend> backend;
   if (selected_backend == os_crypt::SelectedLinuxBackend::KWALLET ||
       selected_backend == os_crypt::SelectedLinuxBackend::KWALLET5) {
+#if defined(USE_DBUS)
     VLOG(1) << "Trying KWallet for password storage.";
     base::nix::DesktopEnvironment used_desktop_env =
         selected_backend == os_crypt::SelectedLinuxBackend::KWALLET
@@ -215,6 +216,7 @@ PasswordStoreFactory::BuildServiceInstanceFor(
     } else {
       backend.reset();
     }
+#endif  // defined(USE_DBUS)
   } else if (selected_backend == os_crypt::SelectedLinuxBackend::GNOME_ANY ||
              selected_backend ==
                  os_crypt::SelectedLinuxBackend::GNOME_KEYRING ||
diff --git chromium-69.0.3497.100/device/bluetooth/bluetooth_adapter_factory.cc
index 8ec9492..fecfb7d 100644
--- chromium-69.0.3497.100/device/bluetooth/bluetooth_adapter_factory.cc
+++ chromium-69.0.3497.100/device/bluetooth/bluetooth_adapter_factory.cc
@@ -76,7 +76,7 @@ bool BluetoothAdapterFactory::IsBluetoothSupported() {
   // instance even on platforms that would otherwise not support it.
   if (default_adapter.Get())
     return true;
-#if defined(OS_ANDROID) || defined(OS_WIN) || defined(OS_LINUX) || \
+#if defined(OS_ANDROID) || defined(OS_WIN) || (defined(OS_LINUX) && defined(USE_DBUS)) || \
     defined(OS_MACOSX)
   return true;
 #else
@@ -99,7 +99,7 @@ bool BluetoothAdapterFactory::IsLowEnergySupported() {
   return base::win::GetVersion() >= base::win::VERSION_WIN10;
 #elif defined(OS_MACOSX)
   return base::mac::IsAtLeastOS10_10();
-#elif defined(OS_LINUX)
+#elif (defined(OS_LINUX) && defined(USE_DBUS))
   return true;
 #else
   return false;
diff --git chromium-69.0.3497.100/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
index f201275..8efdfbe 100644
--- chromium-69.0.3497.100/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
+++ chromium-69.0.3497.100/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
@@ -1326,7 +1326,7 @@ void BluetoothLowEnergyCreateServiceFunction::DoWork() {
 // TODO: Ideally this should be handled by our feature system, so that this
 // code doesn't even compile on OSes it isn't being used on, but currently this
 // is not possible.
-#if !defined(OS_WIN)
+#if !defined(OS_WIN) && (!defined(OS_LINUX) || defined(USE_DBUS))
   base::WeakPtr<device::BluetoothLocalGattService> service =
       device::BluetoothLocalGattService::Create(
           event_router_->adapter(),
@@ -1355,6 +1355,7 @@ bool BluetoothLowEnergyCreateCharacteristicFunction::ParseParams() {
 }
 
 void BluetoothLowEnergyCreateCharacteristicFunction::DoWork() {
+#if !defined(OS_LINUX) || defined(USE_DBUS)
   device::BluetoothLocalGattService* service =
       event_router_->adapter()->GetGattService(params_->service_id);
   if (!service) {
@@ -1375,6 +1376,9 @@ void BluetoothLowEnergyCreateCharacteristicFunction::DoWork() {
 
   Respond(ArgumentList(apibtle::CreateCharacteristic::Results::Create(
       characteristic->GetIdentifier())));
+#else
+  Respond(Error(kErrorPlatformNotSupported));
+#endif
 }
 
 // createDescriptor:
@@ -1391,6 +1395,7 @@ bool BluetoothLowEnergyCreateDescriptorFunction::ParseParams() {
 }
 
 void BluetoothLowEnergyCreateDescriptorFunction::DoWork() {
+#if !defined(OS_LINUX) || defined(USE_DBUS)
   device::BluetoothLocalGattCharacteristic* characteristic =
       event_router_->GetLocalCharacteristic(params_->characteristic_id);
   if (!characteristic) {
@@ -1406,6 +1411,9 @@ void BluetoothLowEnergyCreateDescriptorFunction::DoWork() {
 
   Respond(ArgumentList(
       apibtle::CreateDescriptor::Results::Create(descriptor->GetIdentifier())));
+#else
+  Respond(Error(kErrorPlatformNotSupported));
+#endif
 }
 
 // registerService:
diff --git chromium-69.0.3497.100/services/device/battery/battery_status_manager_default.cc
index 0249c6f..ab2ff4e 100644
--- chromium-69.0.3497.100/services/device/battery/battery_status_manager_default.cc
+++ chromium-69.0.3497.100/services/device/battery/battery_status_manager_default.cc
@@ -22,11 +22,10 @@ class BatteryStatusManagerDefault : public BatteryStatusManager {
  private:
   // BatteryStatusManager:
   bool StartListeningBatteryChange() override {
-    NOTIMPLEMENTED();
     return false;
   }
 
-  void StopListeningBatteryChange() override { NOTIMPLEMENTED(); }
+  void StopListeningBatteryChange() override { }
 
   DISALLOW_COPY_AND_ASSIGN(BatteryStatusManagerDefault);
 };
diff --git chromium-69.0.3497.100/services/device/wake_lock/power_save_blocker/BUILD.gn
index dfce90f..75eaadb 100644
--- chromium-69.0.3497.100/services/device/wake_lock/power_save_blocker/BUILD.gn
+++ chromium-69.0.3497.100/services/device/wake_lock/power_save_blocker/BUILD.gn
@@ -62,6 +62,9 @@ source_set("power_save_blocker") {
   if (is_chromeos || !use_x11 || !use_dbus) {
     sources -= [ "power_save_blocker_x11.cc" ]
   }
+  if (!is_chromeos && use_x11 && !use_dbus) {
+    sources += [ "power_save_blocker_ozone.cc" ]
+  }
 
   if (is_android) {
     deps += [ ":jni_headers" ]
