diff --git chromium-81.0.4044.113/third_party/protobuf/src/google/protobuf/repeated_field.h chromium-81.0.4044.113/third_party/protobuf/src/google/protobuf/repeated_field.h
index b5b193c791..443485462d 100644
--- chromium-81.0.4044.113/third_party/protobuf/src/google/protobuf/repeated_field.h
+++ chromium-81.0.4044.113/third_party/protobuf/src/google/protobuf/repeated_field.h
@@ -804,7 +804,7 @@ class StringTypeHandler {
 // RepeatedPtrField is like RepeatedField, but used for repeated strings or
 // Messages.
 template <typename Element>
-class RepeatedPtrField final : private internal::RepeatedPtrFieldBase {
+class PROTOBUF_EXPORT RepeatedPtrField final : private internal::RepeatedPtrFieldBase {
  public:
   RepeatedPtrField();
   explicit RepeatedPtrField(Arena* arena);
