diff --git chromium-81.0.4044.113/build/config/ui.gni chromium-81.0.4044.113/build/config/ui.gni
index ce69f968f3..a9f5bfad7a 100644
--- chromium-81.0.4044.113/build/config/ui.gni
+++ chromium-81.0.4044.113/build/config/ui.gni
@@ -48,8 +48,10 @@ use_x11 = is_linux && !use_ozone
 # Make sure glib is not used if building for ChromeOS/Chromecast
 assert(!use_glib || (is_linux && !is_chromeos && !is_chromecast))
 
-# Whether to use atk, the Accessibility ToolKit library
-use_atk = is_desktop_linux && use_x11
+declare_args() {
+  # Whether to use atk, the Accessibility ToolKit library
+  use_atk = is_desktop_linux && use_x11
+}
 # =============================================
 #   PLEASE DO NOT ADD MORE FLAGS TO THIS FILE
 # =============================================
