diff --git chromium-60.0.3112.78/chrome/browser/BUILD.gn chromium-60.0.3112.78/chrome/browser/BUILD.gn
index c79b28b..22ad72a 100644
--- chromium-60.0.3112.78/chrome/browser/BUILD.gn
+++ chromium-60.0.3112.78/chrome/browser/BUILD.gn
@@ -3858,14 +3858,18 @@ split_static_library("browser") {
       "first_run/upgrade_util_linux.cc",
       "first_run/upgrade_util_linux.h",
       "icon_loader_auralinux.cc",
-      "password_manager/native_backend_kwallet_x.cc",
-      "password_manager/native_backend_kwallet_x.h",
       "platform_util_linux.cc",
       "shell_integration_linux.cc",
       "shell_integration_linux.h",
       "speech/tts_linux.cc",
       "web_applications/web_app_linux.cc",
     ]
+    if (use_dbus) {
+      sources += [
+        "password_manager/native_backend_kwallet_x.cc",
+        "password_manager/native_backend_kwallet_x.h",
+      ]
+    }
 
     if (enable_native_notifications) {
       sources += [
diff --git chromium-60.0.3112.78/chrome/browser/chrome_browser_main_linux.cc chromium-60.0.3112.78/chrome/browser/chrome_browser_main_linux.cc
index 1fc00fa..caa5719 100644
--- chromium-60.0.3112.78/chrome/browser/chrome_browser_main_linux.cc
+++ chromium-60.0.3112.78/chrome/browser/chrome_browser_main_linux.cc
@@ -82,7 +82,7 @@ void ChromeBrowserMainPartsLinux::PostProfileInit() {
 }
 
 void ChromeBrowserMainPartsLinux::PostMainMessageLoopStart() {
-#if !defined(OS_CHROMEOS)
+#if !defined(OS_CHROMEOS) && defined(USE_DBUS)
   bluez::DBusThreadManagerLinux::Initialize();
   bluez::BluezDBusManager::Initialize(
       bluez::DBusThreadManagerLinux::Get()->GetSystemBus(), false);
@@ -92,7 +92,7 @@ void ChromeBrowserMainPartsLinux::PostMainMessageLoopStart() {
 }
 
 void ChromeBrowserMainPartsLinux::PostDestroyThreads() {
-#if !defined(OS_CHROMEOS)
+#if !defined(OS_CHROMEOS) && defined(USE_DBUS)
   bluez::BluezDBusManager::Shutdown();
   bluez::DBusThreadManagerLinux::Shutdown();
 #endif
diff --git chromium-60.0.3112.78/chrome/browser/password_manager/password_store_factory.cc chromium-60.0.3112.78/chrome/browser/password_manager/password_store_factory.cc
index 84e4f05..03f6127 100644
--- chromium-60.0.3112.78/chrome/browser/password_manager/password_store_factory.cc
+++ chromium-60.0.3112.78/chrome/browser/password_manager/password_store_factory.cc
@@ -200,6 +200,7 @@ PasswordStoreFactory::BuildServiceInstanceFor(
   std::unique_ptr<PasswordStoreX::NativeBackend> backend;
   if (selected_backend == os_crypt::SelectedLinuxBackend::KWALLET ||
       selected_backend == os_crypt::SelectedLinuxBackend::KWALLET5) {
+#if defined(USE_DBUS)
     VLOG(1) << "Trying KWallet for password storage.";
     base::nix::DesktopEnvironment used_desktop_env =
         selected_backend == os_crypt::SelectedLinuxBackend::KWALLET
@@ -212,6 +213,7 @@ PasswordStoreFactory::BuildServiceInstanceFor(
     } else {
       backend.reset();
     }
+#endif  // defined(USE_DBUS)
   } else if (selected_backend == os_crypt::SelectedLinuxBackend::GNOME_ANY ||
              selected_backend ==
                  os_crypt::SelectedLinuxBackend::GNOME_KEYRING ||
diff --git chromium-60.0.3112.78/device/bluetooth/bluetooth_adapter_factory.cc chromium-60.0.3112.78/device/bluetooth/bluetooth_adapter_factory.cc
index cee5e1d..df6a895 100644
--- chromium-60.0.3112.78/device/bluetooth/bluetooth_adapter_factory.cc
+++ chromium-60.0.3112.78/device/bluetooth/bluetooth_adapter_factory.cc
@@ -77,7 +77,7 @@ bool BluetoothAdapterFactory::IsBluetoothSupported() {
   if (default_adapter.Get())
     return true;
-#if defined(OS_ANDROID) || defined(OS_WIN) || defined(OS_LINUX) || \
+#if defined(OS_ANDROID) || defined(OS_WIN) || (defined(OS_LINUX) && defined(USE_DBUS)) || \
     defined(OS_MACOSX)
   return true;
 #else
   return false;
@@ -99,7 +99,7 @@ bool BluetoothAdapterFactory::IsLowEnergySupported() {
   return base::win::GetVersion() >= base::win::VERSION_WIN10;
 #elif defined(OS_MACOSX)
   return base::mac::IsAtLeastOS10_10();
-#elif defined(OS_LINUX)
+#elif (defined(OS_LINUX) && defined(USE_DBUS))
   return true;
 #else
   return false;
diff --git chromium-60.0.3112.78/services/device/wake_lock/power_save_blocker/BUILD.gn chromium-60.0.3112.78/services/device/wake_lock/power_save_blocker/BUILD.gn
index 3d5a127..f8ea3e0 100644
--- chromium-60.0.3112.78/services/device/wake_lock/power_save_blocker/BUILD.gn
+++ chromium-60.0.3112.78/services/device/wake_lock/power_save_blocker/BUILD.gn
@@ -52,6 +52,9 @@ component("power_save_blocker") {
   if (is_chromeos || !use_x11 || !use_dbus) {
     sources -= [ "power_save_blocker_x11.cc" ]
   }
+  if (!is_chromeos && use_x11 && !use_dbus) {
+    sources += [ "power_save_blocker_ozone.cc" ]
+  }
 
   if (is_android) {
     deps += [ ":jni_headers" ]
diff --git chromium-60.0.3112.78/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc chromium-60.0.3112.78/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
index 515838f..59360ec 100644
--- chromium-60.0.3112.78/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
+++ chromium-60.0.3112.78/extensions/browser/api/bluetooth_low_energy/bluetooth_low_energy_api.cc
@@ -1316,7 +1316,7 @@ void BluetoothLowEnergyCreateServiceFunction::DoWork() {
 // TODO: Ideally this should be handled by our feature system, so that this
 // code doesn't even compile on OSes it isn't being used on, but currently this
 // is not possible.
-#if !defined(OS_WIN)
+#if !defined(OS_WIN) && (!defined(OS_LINUX) || defined(USE_DBUS))
   base::WeakPtr<device::BluetoothLocalGattService> service =
       device::BluetoothLocalGattService::Create(
           event_router_->adapter(),
@@ -1337,6 +1337,7 @@ template class BLEPeripheralExtensionFunction<
     apibtle::CreateCharacteristic::Params>;
 
 void BluetoothLowEnergyCreateCharacteristicFunction::DoWork() {
+#if !defined(OS_LINUX) || defined(USE_DBUS)
   device::BluetoothLocalGattService* service =
       event_router_->adapter()->GetGattService(params_->service_id);
   if (!service) {
@@ -1357,6 +1358,9 @@ void BluetoothLowEnergyCreateCharacteristicFunction::DoWork() {
 
   Respond(ArgumentList(apibtle::CreateCharacteristic::Results::Create(
       characteristic->GetIdentifier())));
+#else
+  Respond(Error(kErrorPlatformNotSupported));
+#endif
 }
 
 // createDescriptor:
@@ -1365,6 +1369,7 @@ template class BLEPeripheralExtensionFunction<
     apibtle::CreateDescriptor::Params>;
 
 void BluetoothLowEnergyCreateDescriptorFunction::DoWork() {
+#if !defined(OS_LINUX) || defined(USE_DBUS)
   device::BluetoothLocalGattCharacteristic* characteristic =
       event_router_->GetLocalCharacteristic(params_->characteristic_id);
   if (!characteristic) {
@@ -1380,6 +1385,9 @@ void BluetoothLowEnergyCreateDescriptorFunction::DoWork() {
 
   Respond(ArgumentList(
       apibtle::CreateDescriptor::Results::Create(descriptor->GetIdentifier())));
+#else
+  Respond(Error(kErrorPlatformNotSupported));
+#endif
 }
 
 // registerService:
diff --git chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.cc chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.cc
index b7ba6cd..491ef26 100644
--- chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.cc
+++ chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.cc
@@ -17,10 +17,12 @@
 #include "base/threading/sequenced_worker_pool.h"
 #include "components/onc/onc_constants.h"
 #include "content/public/browser/browser_thread.h"
+#if defined(USE_DBUS)
 #include "dbus/bus.h"
 #include "dbus/message.h"
 #include "dbus/object_path.h"
 #include "dbus/object_proxy.h"
+#endif
 #include "extensions/browser/api/networking_private/network_config_dbus_constants_linux.h"
 #include "extensions/browser/api/networking_private/networking_private_api.h"
 #include "extensions/browser/api/networking_private/networking_private_delegate_observer.h"
@@ -30,6 +32,8 @@
 namespace extensions {
 
 namespace {
+
+#if defined(USE_DBUS)
 // Access Point info strings.
 const char kAccessPointInfoName[] = "Name";
 const char kAccessPointInfoGuid[] = "GUID";
@@ -96,6 +100,7 @@ std::string ConstructNetworkGuid(const dbus::ObjectPath& device_path,
                                  const std::string& ssid) {
   return device_path.value() + "|" + access_point_path.value() + "|" + ssid;
 }
+#endif
 
 // Logs that the method is not implemented and reports |kErrorNotSupported|
 // to the failure callback.
@@ -106,6 +111,7 @@ void ReportNotSupported(
   failure_callback.Run(extensions::networking_private::kErrorNotSupported);
 }
 
+#if defined(USE_DBUS)
 // Fires the appropriate callback when the network connect operation succeeds
 // or fails.
 void OnNetworkConnectOperationCompleted(
@@ -132,9 +138,11 @@ void GetCachedNetworkPropertiesCallback(
   }
   success_callback.Run(std::move(properties));
 }
+#endif
 
 }  // namespace
 
+#if defined(USE_DBUS)
 NetworkingPrivateLinux::NetworkingPrivateLinux()
     : dbus_thread_("Networking Private DBus"), network_manager_proxy_(NULL) {
   base::Thread::Options thread_options(base::MessageLoop::Type::TYPE_IO, 0);
@@ -144,7 +152,12 @@ NetworkingPrivateLinux::NetworkingPrivateLinux()
       FROM_HERE,
       base::Bind(&NetworkingPrivateLinux::Initialize, base::Unretained(this)));
 }
+#else
+NetworkingPrivateLinux::NetworkingPrivateLinux() {
+}
+#endif
 
+#if defined(USE_DBUS)
 NetworkingPrivateLinux::~NetworkingPrivateLinux() {
   dbus_thread_.Stop();
 }
@@ -184,6 +197,7 @@ bool NetworkingPrivateLinux::CheckNetworkManagerSupported(
 
   return true;
 }
+#endif
 
 void NetworkingPrivateLinux::GetProperties(
     const std::string& guid,
@@ -203,6 +217,7 @@ void NetworkingPrivateLinux::GetState(
     const std::string& guid,
     const DictionaryCallback& success_callback,
     const FailureCallback& failure_callback) {
+#if defined(USE_DBUS)
   if (!CheckNetworkManagerSupported(failure_callback))
     return;
 
@@ -219,8 +234,12 @@ void NetworkingPrivateLinux::GetState(
       base::Bind(&GetCachedNetworkPropertiesCallback, base::Passed(&error),
                  base::Passed(&network_properties), success_callback,
                  failure_callback));
+#else
+  ReportNotSupported("NetworkManager over DBus", failure_callback);
+#endif
 }
 
+#if defined(USE_DBUS)
 void NetworkingPrivateLinux::GetCachedNetworkProperties(
     const std::string& guid,
     base::DictionaryValue* properties,
@@ -247,6 +266,7 @@ void NetworkingPrivateLinux::GetCachedNetworkProperties(
   // Swap the new copy into the dictionary that is shared with the reply.
   properties->Swap(temp_properties.get());
 }
+#endif
 
 void NetworkingPrivateLinux::SetProperties(
     const std::string& guid,
@@ -281,6 +301,7 @@ void NetworkingPrivateLinux::GetNetworks(
     int limit,
     const NetworkListCallback& success_callback,
     const FailureCallback& failure_callback) {
+#if defined(USE_DBUS)
   if (!CheckNetworkManagerSupported(failure_callback)) {
     return;
   }
@@ -306,8 +327,12 @@ void NetworkingPrivateLinux::GetNetworks(
       base::Bind(&NetworkingPrivateLinux::OnAccessPointsFound,
                  base::Unretained(this), base::Passed(&network_map),
                  success_callback, failure_callback));
+#else
+  ReportNotSupported("NetworkManager over DBus", failure_callback);
+#endif
 }
 
+#if defined(USE_DBUS)
 bool NetworkingPrivateLinux::GetNetworksForScanRequest() {
   if (!network_manager_proxy_) {
     return false;
@@ -493,11 +518,13 @@ void NetworkingPrivateLinux::DisconnectFromNetwork(const std::string& guid,
     *error = "Failed to disconnect network";
   }
 }
+#endif
 
 void NetworkingPrivateLinux::StartConnect(
     const std::string& guid,
     const VoidCallback& success_callback,
     const FailureCallback& failure_callback) {
+#if defined(USE_DBUS)
   if (!CheckNetworkManagerSupported(failure_callback))
     return;
 
@@ -510,12 +537,16 @@ void NetworkingPrivateLinux::StartConnect(
                  base::Unretained(this), guid, base::Unretained(error.get())),
       base::Bind(&OnNetworkConnectOperationCompleted, base::Passed(&error),
                  success_callback, failure_callback));
+#else
+  ReportNotSupported("NetworkManager over DBus", failure_callback);
+#endif
 }
 
 void NetworkingPrivateLinux::StartDisconnect(
     const std::string& guid,
     const VoidCallback& success_callback,
     const FailureCallback& failure_callback) {
+#if defined(USE_DBUS)
   if (!CheckNetworkManagerSupported(failure_callback))
     return;
 
@@ -528,6 +559,9 @@ void NetworkingPrivateLinux::StartDisconnect(
                  base::Unretained(this), guid, base::Unretained(error.get())),
       base::Bind(&OnNetworkConnectOperationCompleted, base::Passed(&error),
                  success_callback, failure_callback));
+#else
+  ReportNotSupported("NetworkManager over DBus", failure_callback);
+#endif
 }
 
 void NetworkingPrivateLinux::SetWifiTDLSEnabledState(
@@ -608,19 +642,28 @@ bool NetworkingPrivateLinux::DisableNetworkType(const std::string& type) {
 }
 
 bool NetworkingPrivateLinux::RequestScan(const std::string& /* type */) {
+#if defined(USE_DBUS)
   return GetNetworksForScanRequest();
+#else
+  return false;
+#endif
 }
 
 void NetworkingPrivateLinux::AddObserver(
     NetworkingPrivateDelegateObserver* observer) {
+#if defined(USE_DBUS)
   network_events_observers_.AddObserver(observer);
+#endif
 }
 
 void NetworkingPrivateLinux::RemoveObserver(
     NetworkingPrivateDelegateObserver* observer) {
+#if defined(USE_DBUS)
   network_events_observers_.RemoveObserver(observer);
+#endif
 }
 
+#if defined(USE_DBUS)
 void NetworkingPrivateLinux::OnAccessPointsFound(
     std::unique_ptr<NetworkMap> network_map,
     const NetworkListCallback& success_callback,
@@ -1212,5 +1255,6 @@ void NetworkingPrivateLinux::OnNetworksChangedEventTask(
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   OnNetworksChangedEventOnUIThread(*guid_list);
 }
+#endif
 
 }  // namespace extensions
diff --git chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.h chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.h
index ff3c5e5..adbee0b 100644
--- chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.h
+++ chromium-60.0.3112.78/extensions/browser/api/networking_private/networking_private_linux.h
@@ -16,22 +16,26 @@
 #include "components/keyed_service/core/keyed_service.h"
 #include "extensions/browser/api/networking_private/networking_private_delegate.h"
 
+#if defined(USE_DBUS)
 namespace dbus {
 class Bus;
 class ObjectPath;
 class ObjectProxy;
 class Response;
 };
+#endif
 
 namespace extensions {
 
 // Linux NetworkingPrivateDelegate implementation.
 class NetworkingPrivateLinux : public NetworkingPrivateDelegate {
  public:
+#if defined(USE_DBUS)
   using NetworkMap =
       std::map<base::string16, std::unique_ptr<base::DictionaryValue>>;
 
   typedef std::vector<std::string> GuidList;
+#endif
 
   NetworkingPrivateLinux();
 
@@ -103,6 +107,7 @@ class NetworkingPrivateLinux : public NetworkingPrivateDelegate {
   void RemoveObserver(NetworkingPrivateDelegateObserver* observer) override;
 
  private:
+#if defined(USE_DBUS)
   ~NetworkingPrivateLinux() override;
 
   // https://developer.gnome.org/NetworkManager/unstable/spec.html#type-NM_DEVICE_TYPE
@@ -270,6 +275,7 @@ class NetworkingPrivateLinux : public NetworkingPrivateDelegate {
   // Observers to Network Events.
   base::ObserverList<NetworkingPrivateDelegateObserver>
       network_events_observers_;
+#endif
 
   DISALLOW_COPY_AND_ASSIGN(NetworkingPrivateLinux);
 };
diff --git chromium-60.0.3112.78/services/device/battery/battery_status_manager_default.cc chromium-60.0.3112.78/services/device/battery/battery_status_manager_default.cc
index 5d64b34..a224c74 100644
--- chromium-60.0.3112.78/services/device/battery/battery_status_manager_default.cc
+++ chromium-60.0.3112.78/services/device/battery/battery_status_manager_default.cc
@@ -22,11 +22,10 @@ class BatteryStatusManagerDefault : public BatteryStatusManager {
  private:
   // BatteryStatusManager:
   bool StartListeningBatteryChange() override {
-    NOTIMPLEMENTED();
     return false;
   }
 
-  void StopListeningBatteryChange() override { NOTIMPLEMENTED(); }
+  void StopListeningBatteryChange() override { }
 
   DISALLOW_COPY_AND_ASSIGN(BatteryStatusManagerDefault);
 };
