diff --git chromium-72.0.3626.121/components/url_formatter/url_formatter.cc
index b060780..01a249f 100644
--- chromium-72.0.3626.121/components/url_formatter/url_formatter.cc.orig
+++ chromium-72.0.3626.121/components/url_formatter/url_formatter.cc
@@ -410,13 +410,13 @@ IDNConversionStatus IDNToUnicodeOneComponent(const base::char16* comp,
 }  // namespace
 
 const FormatUrlType kFormatUrlOmitNothing = 0;
-const FormatUrlType kFormatUrlOmitUsernamePassword = 1 << 0;
-const FormatUrlType kFormatUrlOmitHTTP = 1 << 1;
-const FormatUrlType kFormatUrlOmitTrailingSlashOnBareHostname = 1 << 2;
-const FormatUrlType kFormatUrlOmitHTTPS = 1 << 3;
-const FormatUrlType kFormatUrlOmitTrivialSubdomains = 1 << 5;
-const FormatUrlType kFormatUrlTrimAfterHost = 1 << 6;
-const FormatUrlType kFormatUrlOmitFileScheme = 1 << 7;
+const FormatUrlType kFormatUrlOmitUsernamePassword = 0;
+const FormatUrlType kFormatUrlOmitHTTP = 0;
+const FormatUrlType kFormatUrlOmitTrailingSlashOnBareHostname = 0;
+const FormatUrlType kFormatUrlOmitHTTPS = 0;
+const FormatUrlType kFormatUrlOmitTrivialSubdomains = 0;
+const FormatUrlType kFormatUrlTrimAfterHost = 0;
+const FormatUrlType kFormatUrlOmitFileScheme = 0;
 
 const FormatUrlType kFormatUrlOmitDefaults =
     kFormatUrlOmitUsernamePassword | kFormatUrlOmitHTTP |
