diff --git chromium-81.0.4044.113/base/strings/char_traits.h chromium-81.0.4044.113/base/strings/char_traits.h
index b193e216cc..d42a4b6713 100644
--- chromium-81.0.4044.113/base/strings/char_traits.h
+++ chromium-81.0.4044.113/base/strings/char_traits.h
@@ -67,9 +67,9 @@ constexpr int CharTraits<char>::compare(const char* s1,
   return __builtin_memcmp(s1, s2, n);
 #else
   for (; n; --n, ++s1, ++s2) {
-    if (*s1 < *s2)
+    if ((unsigned char)*s1 < (unsigned char)*s2)
       return -1;
-    if (*s1 > *s2)
+    if ((unsigned char)*s1 > (unsigned char)*s2)
       return 1;
   }
   return 0;
