diff --git chromium-85.0.4183.83/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc chromium-85.0.4183.83/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
index 28d6e84713..8aa357244a 100644
--- chromium-85.0.4183.83/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
+++ chromium-85.0.4183.83/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
@@ -49,7 +49,9 @@
 #endif  // defined(OS_ANDROID) && defined(__arm__)
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(__GLIBC__)
 #include <gnu/libc-version.h>
+#endif  // defined(__GLIBC__)
 
 #include "base/linux_util.h"
 #include "base/strings/string_split.h"
@@ -291,7 +293,7 @@ void RecordLinuxDistro() {
 #endif  // defined(OS_LINUX) && !defined(OS_CHROMEOS)
 
 void RecordLinuxGlibcVersion() {
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(OS_LINUX) && !defined(OS_CHROMEOS) && defined(__GLIBC__)
   base::Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
diff --git chromium-85.0.4183.83/mojo/public/cpp/platform/named_platform_channel_posix.cc chromium-85.0.4183.83/mojo/public/cpp/platform/named_platform_channel_posix.cc
index 9082ac4da5..cbedae6c3c 100644
--- chromium-85.0.4183.83/mojo/public/cpp/platform/named_platform_channel_posix.cc
+++ chromium-85.0.4183.83/mojo/public/cpp/platform/named_platform_channel_posix.cc
@@ -5,6 +5,7 @@
 #include "mojo/public/cpp/platform/named_platform_channel.h"
 
 #include <errno.h>
+#include <string.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <unistd.h>
