diff --git chromium-54.0.2840.34/base/threading/platform_thread_linux.cc.orig chromium-fixed/base/threading/platform_thread_linux.cc
index 299ae2b..8c0cdaa 100644
--- chromium-54.0.2840.34/base/threading/platform_thread_linux.cc.orig
+++ chromium-fixed/base/threading/platform_thread_linux.cc
@@ -155,12 +155,14 @@ void InitThreading() {}
 void TerminateOnThread() {}
 
 size_t GetDefaultThreadStackSize(const pthread_attr_t& attributes) {
-#if !defined(THREAD_SANITIZER)
-  return 0;
-#else
+#if defined(THREAD_SANITIZER)
   // ThreadSanitizer bloats the stack heavily. Evidence has been that the
   // default stack size isn't enough for some browser tests.
   return 2 * (1 << 23);  // 2 times 8192K (the default stack size on Linux).
+#elif defined(__GLIBC__)
+  return 0;
+#else
+  return 1 * (1 << 23);  // 8192K (the default stack size on glibc)
 #endif
 }
 
diff --git chromium-58.0.3029.81/chrome/app/shutdown_signal_handlers_posix.cc
index 5c7ffd9..c2ffff9 100644
--- chromium-58.0.3029.81/chrome/app/shutdown_signal_handlers_posix.cc.orig
+++ chromium-58.0.3029.81/chrome/app/shutdown_signal_handlers_posix.cc
@@ -183,6 +183,7 @@ void InstallShutdownSignalHandlers(
   g_pipe_pid = getpid();
   g_shutdown_pipe_read_fd = pipefd[0];
   g_shutdown_pipe_write_fd = pipefd[1];
+#ifdef __GLIBC__
 #if !defined(ADDRESS_SANITIZER) && !defined(KEEP_SHADOW_STACKS)
   const size_t kShutdownDetectorThreadStackSize = PTHREAD_STACK_MIN * 2;
 #else
@@ -191,6 +192,12 @@ void InstallShutdownSignalHandlers(
   // size to avoid hitting the guard page.
   const size_t kShutdownDetectorThreadStackSize = PTHREAD_STACK_MIN * 4;
 #endif
+#else
+  // PTHREAD_STACK_MIN is much smaller on musl. ConstructTlsVector uses
+  // 6616 bytes of stack by itself. This matches the value of the GLIBC
+  // minumum * 2.
+  const size_t kShutdownDetectorThreadStackSize = 32768;
+#endif
   ShutdownDetector* detector = new ShutdownDetector(
       g_shutdown_pipe_read_fd, shutdown_callback, task_runner);
   // PlatformThread does not delete its delegate.
