diff --git chromium-61.0.3163.79/net/dns/dns_config_service_posix.cc
index 03d0255..2cb06d7 100644
--- chromium-61.0.3163.79/net/dns/dns_config_service_posix.cc.orig
+++ chromium-61.0.3163.79/net/dns/dns_config_service_posix.cc
@@ -122,7 +122,7 @@ ConfigParsePosixResult ReadDnsConfig(DnsConfig* config) {
   ConfigParsePosixResult result;
   config->unhandled_options = false;
 // TODO(fuchsia): Use res_ninit() when it's implemented on Fuchsia.
-#if defined(OS_OPENBSD) || defined(OS_FUCHSIA)
+#if defined(OS_OPENBSD) || defined(OS_FUCHSIA) || (defined(OS_LINUX) && !defined(__GLIBC__))
   // Note: res_ninit in glibc always returns 0 and sets RES_INIT.
   // res_init behaves the same way.
   memset(&_res, 0, sizeof(_res));
@@ -145,7 +145,7 @@ ConfigParsePosixResult ReadDnsConfig(DnsConfig* config) {
 #else
   res_nclose(&res);
 #endif  // defined(OS_MACOSX) || defined(OS_FREEBSD)
-#endif  // defined(OS_OPENBSD)
+#endif  // defined(OS_OPENBSD) || defined(OS_FUCHSIA) || (defined(OS_LINUX) && !defined(__GLIBC__))
 
 #if defined(OS_MACOSX) && !defined(OS_IOS)
   ConfigParsePosixResult error = DnsConfigWatcher::CheckDnsConfig();
diff --git chromium-54.0.2840.59/net/dns/dns_reloader.cc.orig chromium-54.0.2840.59/net/dns/dns_reloader.cc
index 74534e6..9ca2619 100644
--- chromium-54.0.2840.59/net/dns/dns_reloader.cc.orig
+++ chromium-54.0.2840.59/net/dns/dns_reloader.cc
@@ -4,8 +4,7 @@
 
 #include "net/dns/dns_reloader.h"
 
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID) && !defined(OS_FUCHSIA)
+#if defined(__GLIBC__)
 
 #include <resolv.h>
 
@@ -121,5 +120,4 @@ void DnsReloaderMaybeReload() {
 
 }  // namespace net
 
-#endif  // defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) &&
-        // !defined(OS_ANDROID)
+#endif  // defined(__GLIBC__)
diff --git chromium-54.0.2840.59/net/dns/host_resolver_impl.cc.orig chromium-54.0.2840.59/net/dns/host_resolver_impl.cc
index 8be7731..8412824 100644
--- chromium-54.0.2840.59/net/dns/host_resolver_impl.cc.orig
+++ chromium-54.0.2840.59/net/dns/host_resolver_impl.cc
@@ -2015,8 +2015,7 @@ HostResolverImpl::HostResolverImpl(
   NetworkChangeNotifier::AddIPAddressObserver(this);
   NetworkChangeNotifier::AddConnectionTypeObserver(this);
   NetworkChangeNotifier::AddDNSObserver(this);
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID)
+#if defined(__GLIBC__)
   EnsureDnsReloaderInit();
 #endif
 
diff --git chromium-54.0.2840.59/net/dns/host_resolver_proc.cc.orig chromium-54.0.2840.59/net/dns/host_resolver_proc.cc
index 4bed222..90bead6 100644
--- chromium-54.0.2840.59/net/dns/host_resolver_proc.cc.orig
+++ chromium-54.0.2840.59/net/dns/host_resolver_proc.cc
@@ -193,8 +193,7 @@ int SystemHostResolverCall(const std::string& host,
   // Restrict result set to only this socket type to avoid duplicates.
   hints.ai_socktype = SOCK_STREAM;
 
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID) && !defined(OS_FUCHSIA)
+#if defined(__GLIBC__)
   DnsReloaderMaybeReload();
 #endif
   int err = getaddrinfo(host.c_str(), NULL, &hints, &ai);
