diff --git chromium-54.0.2840.34/tools/gn/bootstrap/bootstrap.py chromium-54.0.2840.34/tools/gn/bootstrap/bootstrap.py
index b07fd0a..b870e7b 100755
--- chromium-54.0.2840.34/tools/gn/bootstrap/bootstrap.py
+++ chromium-54.0.2840.34/tools/gn/bootstrap/bootstrap.py
@@ -159,7 +159,7 @@ def build_gn_with_ninja_manually(tempdir, options):
   mkdir_p(root_gen_dir)
 
   write_buildflag_header_manually(root_gen_dir, 'base/allocator/features.h',
-      {'USE_EXPERIMENTAL_ALLOCATOR_SHIM': 'true' if is_linux else 'false'})
+      {'USE_EXPERIMENTAL_ALLOCATOR_SHIM': 'false'})
 
   write_buildflag_header_manually(root_gen_dir, 'base/debug/debugging_flags.h',
       {'ENABLE_PROFILING': 'false'})
@@ -366,8 +366,6 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/third_party/superfasthash/superfasthash.c',
   ])
   static_libraries['base']['sources'].extend([
-      'base/allocator/allocator_check.cc',
-      'base/allocator/allocator_extension.cc',
       'base/at_exit.cc',
       'base/base_paths.cc',
       'base/base_switches.cc',
@@ -573,8 +571,6 @@ def write_gn_ninja(path, root_gen_dir, options):
         'tool': 'cxx',
     }
     static_libraries['base']['sources'].extend([
-        'base/allocator/allocator_shim.cc',
-        'base/allocator/allocator_shim_default_dispatch_to_glibc.cc',
         'base/memory/shared_memory_posix.cc',
         'base/nix/xdg_util.cc',
         'base/process/internal_linux.cc',
