diff --git chromium-81.0.4044.113/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc chromium-81.0.4044.113/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
index 149b7753c7..a47685d972 100644
--- chromium-81.0.4044.113/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
+++ chromium-81.0.4044.113/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
@@ -46,7 +46,9 @@
 #endif  // defined(OS_ANDROID) && defined(__arm__)
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(__GLIBC__)
 #include <gnu/libc-version.h>
+#endif  // defined(__GLIBC__)
 
 #include "base/linux_util.h"
 #include "base/strings/string_split.h"
@@ -365,7 +367,7 @@ void RecordLinuxDistro() {
 #endif  // defined(OS_LINUX) && !defined(OS_CHROMEOS)
 
 void RecordLinuxGlibcVersion() {
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(OS_LINUX) && !defined(OS_CHROMEOS) && defined(__GLIBC__)
   base::Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
