diff --git chromium-60.0.3112.78/third_party/angle/src/vulkan_support/BUILD.gn chromium-60.0.3112.78/third_party/angle/src/vulkan_support/BUILD.gn
index 51c8c10..3733ed1 100644
--- chromium-60.0.3112.78/third_party/angle/src/vulkan_support/BUILD.gn
+++ chromium-60.0.3112.78/third_party/angle/src/vulkan_support/BUILD.gn
@@ -203,10 +203,6 @@ config("vulkan_loader_config") {
   if (is_win) {
     cflags = [ "/wd4201" ]
   }
-  if (is_linux) {
-    # assume secure_getenv() is available
-    defines += [ "HAVE_SECURE_GETENV" ]
-  }
 }
 
 static_library("vulkan_loader") {
