diff --git chromium-54.0.2840.59/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc.orig chromium-54.0.2840.59/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
index b1af95d..9e7a95f 100644
--- chromium-54.0.2840.59/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc.orig
+++ chromium-54.0.2840.59/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
@@ -36,14 +36,14 @@
 #include <cpu-features.h>
 #endif  // defined(OS_ANDROID) && defined(__arm__)
 
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(OS_LINUX) && defined(__GLIBC__)
 #include <gnu/libc-version.h>
 
 #include "base/version.h"
 #if defined(USE_X11)
 #include "ui/base/x/x11_util.h"
 #endif
-#endif  // defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#endif  // defined(OS_LINUX) && defined(__GLIBC__)
 
 #if defined(USE_OZONE) || defined(USE_X11)
 #include "ui/events/devices/input_device_event_observer.h"
@@ -169,7 +169,7 @@ void RecordStartupMetricsOnBlockingPool() {
 }
 
 void RecordLinuxGlibcVersion() {
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(OS_LINUX) && defined(__GLIBC__)
   Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
@@ -193,7 +193,7 @@ void RecordLinuxGlibcVersion() {
 #endif
 }
 
-#if defined(USE_X11) && !defined(OS_CHROMEOS)
+#if defined(USE_X11) && defined(__GLIBC__)
 UMALinuxWindowManager GetLinuxWindowManager() {
   switch (ui::GuessWindowManager()) {
     case ui::WM_UNKNOWN:
@@ -326,7 +326,7 @@ void ChromeBrowserMainExtraPartsMetrics::PreBrowserStart() {
 
 void ChromeBrowserMainExtraPartsMetrics::PostBrowserStart() {
   RecordLinuxGlibcVersion();
-#if defined(USE_X11) && !defined(OS_CHROMEOS)
+#if defined(USE_X11) && defined(__GLIBC__)
   UMA_HISTOGRAM_ENUMERATION("Linux.WindowManager",
                             GetLinuxWindowManager(),
                             UMA_LINUX_WINDOW_MANAGER_COUNT);
diff --git chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc.orig chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc
index 59582c5..65673c6 100644
--- chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc.orig
+++ chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc
@@ -4,6 +4,7 @@
 
 #include "device/serial/serial_io_handler_posix.h"
 
+#include <asm/ioctls.h>
 #include <sys/ioctl.h>
 #include <termios.h>
 
