diff --git chromium-81.0.4044.113/third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h chromium-81.0.4044.113/third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h
index d03c7a88d3..d41f3d716c 100644
--- chromium-81.0.4044.113/third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h
+++ chromium-81.0.4044.113/third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h
@@ -36,6 +36,7 @@
 #include <elf.h>
 #include <link.h>
 #include <stddef.h>
+#include <limits.h>
 
 #include "common/memory_range.h"
 
@@ -51,9 +52,9 @@ class ElfCoreDump {
   typedef ElfW(Phdr) Phdr;
   typedef ElfW(Word) Word;
   typedef ElfW(Addr) Addr;
-#if __WORDSIZE == 32
+#if LONG_BIT == 32
   static const int kClass = ELFCLASS32;
-#elif __WORDSIZE == 64
+#elif LONG_BIT == 64
   static const int kClass = ELFCLASS64;
 #else
 #error "Unsupported __WORDSIZE for ElfCoreDump."
