diff --git chromium-73.0.3683.103/base/native_library_posix.cc chromium-73.0.3683.103/base/native_library_posix.cc
index c886d47..f80638d 100644
--- chromium-73.0.3683.103/base/native_library_posix.cc
+++ chromium-73.0.3683.103/base/native_library_posix.cc
@@ -34,7 +34,7 @@ NativeLibrary LoadNativeLibraryWithOptions(const FilePath& library_path,
   // further investigation, as it might vary across versions. Crash here to
   // warn developers that they're trying to rely on uncertain behavior.
   CHECK(!options.prefer_own_symbols);
-#else
+#elif defined(RTLD_DEEPBIND)
   if (options.prefer_own_symbols)
     flags |= RTLD_DEEPBIND;
 #endif
