diff --git chromium-57.0.2987.98/third_party/WebKit/Source/platform/wtf/StackUtil.cpp
index f9e51b8..2b4c6b9 100644
--- chromium-57.0.2987.98/third_party/WebKit/Source/platform/wtf/StackUtil.cpp.orig
+++ chromium-57.0.2987.98/third_party/WebKit/Source/platform/wtf/StackUtil.cpp
@@ -28,7 +28,7 @@ size_t getUnderestimatedStackSize() {
 // FIXME: On Mac OSX and Linux, this method cannot estimate stack size
 // correctly for the main thread.
 
-#if defined(__GLIBC__) || OS(ANDROID) || OS(FREEBSD)
+#if OS(LINUX) || OS(ANDROID) || OS(FREEBSD)
   // pthread_getattr_np() can fail if the thread is not invoked by
   // pthread_create() (e.g., the main thread of webkit_unit_tests).
   // If so, a conservative size estimate is returned.
@@ -53,6 +53,9 @@ size_t getUnderestimatedStackSize() {
   pthread_attr_destroy(&attr);
 #endif
 
+#if OS(LINUX) && !defined(__GLIBC__)
+  return 0;
+#else
   // Return a 512k stack size, (conservatively) assuming the following:
   //  - that size is much lower than the pthreads default (x86 pthreads has a 2M
   //    default.)
@@ -60,6 +63,7 @@ size_t getUnderestimatedStackSize() {
   //    low as 512k.
   //
   return 512 * 1024;
+#endif
 #elif OS(MACOSX)
   // pthread_get_stacksize_np() returns too low a value for the main thread on
   // OSX 10.9,
@@ -95,7 +99,7 @@ size_t getUnderestimatedStackSize() {
 }
 
 void* GetStackStart() {
-#if defined(__GLIBC__) || OS(ANDROID) || OS(FREEBSD)
+#if OS(LINUX) || OS(ANDROID) || OS(FREEBSD)
   pthread_attr_t attr;
   int error;
 #if OS(FREEBSD)
