diff --git chromium-54.0.2840.34/base/allocator/allocator_shim_internals.h chromium-fixed/base/allocator/allocator_shim_internals.h
--- chromium-54.0.2840.34/base/allocator/allocator_shim_internals.h
+++ chromium-fixed/base/allocator/allocator_shim_internals.h
@@ -7,7 +7,9 @@
 
 #if defined(__GNUC__)
 
+#if defined(__GLIBC__)
 #include <sys/cdefs.h>  // for __THROW
+#endif
 
 #ifndef __THROW  // Not a glibc system
 #ifdef _NOEXCEPT  // LLVM libc++ uses noexcept instead
diff --git chromium-54.0.2840.34/sandbox/linux/suid/sandbox.c.orig chromium-fixed/sandbox/linux/suid/sandbox.c
index b655d1c..84a9f54 100644
--- chromium-54.0.2840.34/sandbox/linux/suid/sandbox.c.orig
+++ chromium-fixed/sandbox/linux/suid/sandbox.c
@@ -42,7 +42,16 @@
 
 static bool DropRoot();
 
+#if defined(TEMP_FAILURE_RETRY)
 #define HANDLE_EINTR(x) TEMP_FAILURE_RETRY(x)
+#else
+#define HANDLE_EINTR(expression) \
+  (__extension__ \
+   ({ long __result; \
+    do __result = (long) (expression); \
+    while (__result == -1L && errno == EINTR); \
+    __result; }))
+#endif
 
 static void FatalError(const char* msg, ...)
     __attribute__((noreturn, format(printf, 1, 2)));
