diff --git chromium-73.0.3683.103/base/debug/stack_trace.cc chromium-73.0.3683.103/base/debug/stack_trace.cc
index ed531f6..36ef6e6 100644
--- chromium-73.0.3683.103/base/debug/stack_trace.cc
+++ chromium-73.0.3683.103/base/debug/stack_trace.cc
@@ -225,7 +225,9 @@ void StackTrace::Print() const {
 }
 
 void StackTrace::OutputToStream(std::ostream* os) const {
+#if defined(__GLIBC__) && !defined(__UCLIBC__) && !defined(_AIX)
   OutputToStreamWithPrefix(os, nullptr);
+#endif
 }
 
 std::string StackTrace::ToString() const {
@@ -233,7 +235,7 @@ std::string StackTrace::ToString() const {
 }
 std::string StackTrace::ToStringWithPrefix(const char* prefix_string) const {
   std::stringstream stream;
-#if !defined(__UCLIBC__) && !defined(_AIX)
+#if defined(__GLIBC__) && !defined(__UCLIBC__) && !defined(_AIX)
   OutputToStreamWithPrefix(&stream, prefix_string);
 #endif
   return stream.str();
diff --git chromium-73.0.3683.103/base/logging.cc chromium-73.0.3683.103/base/logging.cc
index d473555..1f343df 100644
--- chromium-73.0.3683.103/base/logging.cc
+++ chromium-73.0.3683.103/base/logging.cc
@@ -584,8 +584,8 @@ LogMessage::LogMessage(const char* file, int line, LogSeverity severity,
 
 LogMessage::~LogMessage() {
   size_t stack_start = stream_.tellp();
-#if !defined(OFFICIAL_BUILD) && !defined(OS_NACL) && !defined(__UCLIBC__) && \
-    !defined(OS_AIX)
+#if !defined(OFFICIAL_BUILD) && !defined(OS_NACL) && defined(__GLIBC__) && \
+    !defined(__UCLIBC__) && !defined(OS_AIX)
   if (severity_ == LOG_FATAL && !base::debug::BeingDebugged()) {
     // Include a stack trace on a fatal, unless a debugger is attached.
     base::debug::StackTrace trace;
