diff --git chromium-61.0.3163.79/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
index 9067e84..1ed51bb 100644
--- chromium-61.0.3163.79/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc.orig
+++ chromium-61.0.3163.79/chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc
@@ -39,7 +39,9 @@
 #endif  // defined(OS_ANDROID) && defined(__arm__)
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(__GLIBC__)
 #include <gnu/libc-version.h>
+#endif  // defined(__GLIBC__)
 
 #include "base/linux_util.h"
 #include "base/strings/string_split.h"
@@ -295,7 +297,7 @@ void RecordLinuxDistro() {
 #endif  // defined(OS_LINUX) && !defined(OS_CHROMEOS)
 
 void RecordLinuxGlibcVersion() {
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if defined(OS_LINUX) && !defined(OS_CHROMEOS) && defined(__GLIBC__)
   base::Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
diff --git chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc.orig chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc
index 59582c5..65673c6 100644
--- chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc.orig
+++ chromium-54.0.2840.59/device/serial/serial_io_handler_posix.cc
@@ -4,6 +4,7 @@
 
 #include "device/serial/serial_io_handler_posix.h"
 
+#include <asm/ioctls.h>
 #include <sys/ioctl.h>
 #include <termios.h>
 
