diff --git chromium-66.0.3359.117/third_party/angle/third_party/vulkan-validation-layers/BUILD.gn
index e945079..ad3c43d 100644
--- chromium-66.0.3359.117/third_party/angle/third_party/vulkan-validation-layers/BUILD.gn.orig
+++ chromium-66.0.3359.117/third_party/angle/third_party/vulkan-validation-layers/BUILD.gn
@@ -263,10 +263,6 @@ config("vulkan_loader_config") {
   if (is_win) {
     cflags = [ "/wd4201" ]
   }
-  if (is_linux) {
-    # assume secure_getenv() is available
-    defines += [ "HAVE_SECURE_GETENV" ]
-  }
 }
 
 if (!is_android) {
