diff --git chromium-66.0.3359.117/buildtools/third_party/libc++/trunk/include/__config
index 470b22d..07dae49 100644
--- chromium-66.0.3359.117/buildtools/third_party/libc++/trunk/include/__config.orig
+++ chromium-66.0.3359.117/buildtools/third_party/libc++/trunk/include/__config
@@ -11,6 +11,8 @@
 #ifndef _LIBCPP_CONFIG
 #define _LIBCPP_CONFIG
 
+#define _LIBCPP_HAS_MUSL_LIBC
+
 #if defined(_MSC_VER) && !defined(__clang__)
 #if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
 #define _LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER
