diff --git chromium/breakpad/src/client/linux/handler/exception_handler.cc.orig chromium/breakpad/src/client/linux/handler/exception_handler.cc
index b63f973..bf99823 100644
--- chromium/breakpad/src/client/linux/handler/exception_handler.cc.orig
+++ chromium/breakpad/src/client/linux/handler/exception_handler.cc
@@ -477,7 +477,9 @@ bool ExceptionHandler::SimulateSignalDelivery(int sig) {
   siginfo.si_code = SI_USER;
   siginfo.si_pid = getpid();
   struct ucontext context;
+#if defined(__GLIBC__)
   getcontext(&context);
+#endif
   return HandleSignal(sig, &siginfo, &context);
 }
 
@@ -647,8 +649,10 @@ bool ExceptionHandler::WriteMinidump() {
   sys_prctl(PR_SET_DUMPABLE, 1, 0, 0, 0);
 
   CrashContext context;
+#if defined(__GLIBC__)
   int getcontext_result = getcontext(&context.context);
   if (getcontext_result)
+#endif
     return false;
 
 #if defined(__i386__)
