diff --git chromium-85.0.4183.83/buildtools/third_party/libc++/trunk/include/__config chromium-85.0.4183.83/buildtools/third_party/libc++/trunk/include/__config
index 044cd0ceb0..bd73374034 100644
--- chromium-85.0.4183.83/buildtools/third_party/libc++/trunk/include/__config
+++ chromium-85.0.4183.83/buildtools/third_party/libc++/trunk/include/__config
@@ -10,6 +10,8 @@
 #ifndef _LIBCPP_CONFIG
 #define _LIBCPP_CONFIG
 
+#define _LIBCPP_HAS_MUSL_LIBC
+
 #if defined(_MSC_VER) && !defined(__clang__)
 #  if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
 #    define _LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER
