From c005449211872ae81c72a85f90bc5be8aa08f62a Mon Sep 17 00:00:00 2001
From: bcook <>
Date: Sun, 6 Nov 2016 09:28:25 +0000
Subject: [PATCH] use the correct function for free ok beck@

---
 crypto/x509/x509_vpm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/crypto/x509/x509_vpm.c b/crypto/x509/x509_vpm.c
index 565603c..3ab3a55 100644
--- a/crypto/x509/x509_vpm.c
+++ b/crypto/x509/x509_vpm.c
@@ -94,7 +94,7 @@ sk_deep_copy(void *sk_void, void *copy_func_void, void *free_func_void)
 {
 	_STACK *sk = sk_void;
 	void *(*copy_func)(void *) = copy_func_void;
-	void (*free_func)(void *) = copy_func_void;
+	void (*free_func)(void *) = free_func_void;
 	_STACK *ret = sk_dup(sk);
 
 	if (ret == NULL)
