https://github.com/openntpd-portable/openntpd-portable/issues/43
diff --git a/src/client.c b/src/client.c
index 3de5268..5ee7ea8 100644
--- a/src/client.c
+++ b/src/client.c
@@ -402,8 +402,10 @@ client_dispatch(struct ntp_peer *p, u_int8_t settime)
 	    (long long)interval);

 	client_update(p);
-	if (settime)
+	if (settime) {
 		priv_settime(p->reply[p->shift].offset);
+		p->reply[p->shift].good = 0;
+	}

 	if (++p->shift >= OFFSET_ARRAY_SIZE)
 		p->shift = 0;
