From fcc31c4c1fbf0bb89ea95b5896d4d6beceda6566 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sat, 25 Feb 2017 23:01:19 -0600
Subject: [PATCH 4/5] Presence of libraries does not depend on architecture

---
 src/libstd/build.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libstd/build.rs b/src/libstd/build.rs
index 9fb83ad759..a763d9727e 100644
--- a/src/libstd/build.rs
+++ b/src/libstd/build.rs
@@ -30,7 +30,7 @@ fn main() {
             println!("cargo:rustc-link-lib=dl");
             println!("cargo:rustc-link-lib=log");
             println!("cargo:rustc-link-lib=gcc");
-        } else if !target.contains("musl") || target.contains("mips") {
+        } else if !target.contains("musl") {
             println!("cargo:rustc-link-lib=dl");
             println!("cargo:rustc-link-lib=rt");
             println!("cargo:rustc-link-lib=pthread");
-- 
2.11.0

