From 038a71e83757ad0884c797e363d25d2a99943135 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 10 Jan 2018 13:36:41 -0600
Subject: [PATCH 08/14] Don't pass CFLAGS to the C++ compiler

---
 src/bootstrap/builder.rs | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index ec7eebd1d0..46a5ce27da 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -940,7 +940,7 @@ impl<'a> Builder<'a> {
             let cflags = self.cflags(target).join(" ");
             cargo
                 .env(format!("CFLAGS_{}", target), cflags.clone())
-                .env("CFLAGS", cflags.clone());
+                .env("CFLAGS", cflags);
 
             if let Some(ar) = self.ar(target) {
                 let ranlib = format!("{} s", ar.display());
@@ -955,9 +955,7 @@ impl<'a> Builder<'a> {
                 let cxx = ccacheify(&cxx);
                 cargo
                     .env(format!("CXX_{}", target), &cxx)
-                    .env("CXX", &cxx)
-                    .env(format!("CXXFLAGS_{}", target), cflags.clone())
-                    .env("CXXFLAGS", cflags);
+                    .env("CXX", &cxx);
             }
         }
 
-- 
2.16.4

