From 4affbaecfc9706253ce31e0008ad42dcd7f35a29 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Mon, 11 Sep 2017 11:21:56 -0500
Subject: [PATCH 11/25] Add missing OpenSSL configurations for musl targets

---
 src/bootstrap/native.rs | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/bootstrap/native.rs b/src/bootstrap/native.rs
index 424264647f..a0329910be 100644
--- a/src/bootstrap/native.rs
+++ b/src/bootstrap/native.rs
@@ -643,9 +643,13 @@ impl Step for Openssl {
             "arm-linux-androideabi" => "android",
             "arm-unknown-linux-gnueabi" => "linux-armv4",
             "arm-unknown-linux-gnueabihf" => "linux-armv4",
+            "arm-unknown-linux-musleabi" => "linux-armv4",
+            "arm-unknown-linux-musleabihf" => "linux-armv4",
+            "armv5te-unknown-linux-musleabi" => "linux-armv4",
             "armv6-unknown-netbsd-eabihf" => "BSD-generic32",
             "armv7-linux-androideabi" => "android-armv7",
             "armv7-unknown-linux-gnueabihf" => "linux-armv4",
+            "armv7-unknown-linux-musleabihf" => "linux-armv4",
             "armv7-unknown-netbsd-eabihf" => "BSD-generic32",
             "i586-unknown-linux-gnu" => "linux-elf",
             "i586-unknown-linux-musl" => "linux-elf",
@@ -656,13 +660,17 @@ impl Step for Openssl {
             "i686-unknown-linux-musl" => "linux-elf",
             "i686-unknown-netbsd" => "BSD-x86-elf",
             "mips-unknown-linux-gnu" => "linux-mips32",
+            "mips-unknown-linux-musl" => "linux-mips32",
             "mips64-unknown-linux-gnuabi64" => "linux64-mips64",
             "mips64el-unknown-linux-gnuabi64" => "linux64-mips64",
             "mipsel-unknown-linux-gnu" => "linux-mips32",
+            "mipsel-unknown-linux-musl" => "linux-mips32",
             "powerpc-unknown-linux-gnu" => "linux-ppc",
             "powerpc-unknown-linux-gnuspe" => "linux-ppc",
+            "powerpc-unknown-linux-musl" => "linux-ppc",
             "powerpc-unknown-netbsd" => "BSD-generic32",
             "powerpc64-unknown-linux-gnu" => "linux-ppc64",
+            "powerpc64-unknown-linux-musl" => "linux-ppc64",
             "powerpc64le-unknown-linux-gnu" => "linux-ppc64le",
             "powerpc64le-unknown-linux-musl" => "linux-ppc64le",
             "s390x-unknown-linux-gnu" => "linux64-s390x",
@@ -690,6 +698,10 @@ impl Step for Openssl {
             configure.arg("-mandroid");
             configure.arg("-fomit-frame-pointer");
         }
+        // OpenSSL ships incompatible ELFv1 ABI assembly code
+        if target == "powerpc64-unknown-linux-musl" {
+            configure.arg("no-asm");
+        }
         if target == "sparc64-unknown-netbsd" {
             // Need -m64 to get assembly generated correctly for sparc64.
             configure.arg("-m64");
-- 
2.19.1

