From ca7e83bc78d65d9ae279a159dcd3e37258e252f4 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Mon, 10 Sep 2018 01:36:00 +0000
Subject: [PATCH 17/25] x.py: Use python3 instead of python

---
 x.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x.py b/x.py
index e277ab98be..a75176a3f9 100755
--- a/x.py
+++ b/x.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2016 The Rust Project Developers. See the COPYRIGHT
 # file at the top-level directory of this distribution and at
 # http://rust-lang.org/COPYRIGHT.
-- 
2.19.1

