--- rustc-1.38.0-src/src/llvm-project/llvm/lib/Target/PowerPC/PPCISelLowering.cpp
+++ rustc-1.38.0-src/src/llvm-project/llvm/lib/Target/PowerPC/PPCISelLowering.cpp
@@ -4952,7 +4952,8 @@ PrepareCall(SelectionDAG &DAG, SDValue &Callee, SDValue &InFlag, SDValue &Chain,
   if (auto *G = dyn_cast<GlobalAddressSDNode>(Callee))
     GV = G->getGlobal();
   bool Local = TM.shouldAssumeDSOLocal(*Mod, GV);
-  bool UsePlt = !Local && Subtarget.isTargetELF() && !isPPC64;
+  bool UsePlt = !Local && Subtarget.isTargetELF() && !isPPC64 &&
+                TM.isPositionIndependent();
 
   // If the callee is a GlobalAddress/ExternalSymbol node (quite common,
   // every direct call is) turn it into a TargetGlobalAddress /
