diff --git a/src/bootstrap/lib.rs b/src/bootstrap/lib.rs
--- a/src/bootstrap/lib.rs
+++ b/src/bootstrap/lib.rs
@@ -620,7 +620,7 @@ impl Build {
         // gcc-rs because the build scripts will determine that for themselves.
         let mut base = self.cc[&target].0.args().iter()
                            .map(|s| s.to_string_lossy().into_owned())
-                           .filter(|s| !s.starts_with("-O") && !s.starts_with("/O"))
+                           .filter(|s| !s.starts_with("-O") && !s.starts_with("/O") && !s.starts_with("-static"))
                            .collect::<Vec<_>>();
 
         // If we're compiling on macOS then we add a few unconditional flags
